lib/std/math/pow.zig

```diff
@@ -178,29 +178,37 @@ pub fn pow(comptime T: type, x: T, y: T) T {
     return math.scalbn(a1, ae);
 }
 
-fn isOddInteger(x: f64) bool {
-    if (@abs(x) >= 1 << 53) {
-        // From https://golang.org/src/math/pow.go
-        // 1 << 53 is the largest exact integer in the float64 format.
+fn isOddInteger(comptime T: type, x: T) bool {
+    if (@abs(x) >= 1 << (math.floatFractionalBits(T) + 1)) {
+        // From https://golang.org/src/math/pow.go (adapted for generic type T)
+        // 1 << (floatFractionalBits(T) + 1) is the largest exact integer in the floating point format.
         // Any number outside this range will be truncated before the decimal point and therefore will always be
         // an even integer.
-        // Without this check and if x overflows i64 the @intFromFloat(r.ipart) conversion below will panic
+        // Without this check and if x overflows i128 the @intFromFloat(r.ipart) conversion below will panic
         return false;
     }
     const r = math.modf(x);
-    return r.fpart == 0.0 and @as(i64, @intFromFloat(r.ipart)) & 1 == 1;
+    return r.fpart == 0.0 and @as(i128, @intFromFloat(r.ipart)) & 1 == 1;
 }
 
 test isOddInteger {
-    try expect(isOddInteger(math.maxInt(i64) * 2) == false);
-    try expect(isOddInteger(math.maxInt(i64) * 2 + 1) == false);
-    try expect(isOddInteger(1 << 53) == false);
-    try expect(isOddInteger(12.0) == false);
-    try expect(isOddInteger(15.0) == true);
+    try expect(isOddInteger(f64, math.maxInt(i64) * 2) == false);
+    try expect(isOddInteger(f64, math.maxInt(i64) * 2 + 1) == false);
+    try expect(isOddInteger(f64, 1 << 53) == false);
+    try expect(isOddInteger(f64, 12.0) == false);
+    try expect(isOddInteger(f64, 15.0) == true);
 }
 
 test pow {
     const epsilon = 0.000001;
+    const epsilon16 = 0.01;
+
+    try expect(math.approxEqAbs(f16, pow(f16, 0.0, 3.3), 0.0, epsilon16));
+    try expect(math.approxEqAbs(f16, pow(f16, 0.8923, 3.3), 0.686572, epsilon16));
+    try expect(math.approxEqAbs(f16, pow(f16, 0.2, 3.3), 0.004936, epsilon16));
+    try expect(math.approxEqAbs(f16, pow(f16, 1.5, 3.3), 3.811546, epsilon16));
+    try expect(math.approxEqAbs(f16, pow(f16, 37.45, 3.3), 155736.7160616, epsilon16));
+    try expect(math.approxEqAbs(f16, pow(f16, 89.123, 3.3), 2722490.231436, epsilon16));
```

Comment:
Might be a dumb question, but shouldn't these two result in +inf with the results being way over the maximum value of f16?

Reply:
Good catch! You're right, the expected result for `f16` should be `+inf`.

The test likely passed because the large literal `155736.7...` also overflows to `+inf` when cast to `f16`, causing `approxEqAbs` to hit the `x == y` fast path.

Maybe it's clearer to test explicitly with `== math.inf(f16)` in this overflow case?

Reply:
Oh, I guess that's true. I've mixed up inf comparison and NaN comparison in my mind, but you're right, that's probably what's happening. But I agree that explicitly comparing to inf or even using math.isInf would be more clear here. At least in my opinion.